installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -253,6 +245,35 @@ public static ActionResult InstallStateTool(Session session, out string stateToo
             }
 
             return ActionResult.Success;
+
+        }
+
+        public static ActionResult InstallStateTool(Session session, out string stateToolPath)
+        {
+            RollbarHelper.ConfigureRollbarSingleton(session.CustomActionData["COMMIT_ID"]);
+
+            session.Log("Installing State Tool if necessary");
+            if (session.CustomActionData["STATE_TOOL_INSTALLED"] == "true")
+            {
+                stateToolPath = session.CustomActionData["STATE_TOOL_PATH"];
+                session.Log("State Tool is installed, no installation required");
+                Status.ProgressBar.Increment(session, 1);
+                return ActionResult.Success;
+            }
+
+            Status.ProgressBar.StatusMessage(session, "Installing State Tool...");
+            Status.ProgressBar.Increment(session, 1);
+
+            var ret = _installStateTool(session, out stateToolPath);
+            if (ret == ActionResult.Success)
+            {
+                TrackerSingleton.Instance.TrackEventInBackground(session, "stage", "state-tool", "success");
+            }
+            else if (ret == ActionResult.Failure)
+            {
+                TrackerSingleton.Instance.TrackEventInBackground(session, "stage", "state-tool", "failure");
```

Comment:
Wouldn't this also amount to a "finished" "failure"?

Reply:
All "finished" commands are now triggered after the finishing MSI dialogs are displayed (CustomExitDialog => success, CustomFatalError => failure).